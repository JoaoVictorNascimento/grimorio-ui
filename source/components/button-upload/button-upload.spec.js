import '../../../internals/test/helper';

import ButttonUpload from './button-upload-component';
import Loader from '../loader';
import Icon from '../icon';

const props = {
  disabled: false,
  onChange: jest.fn(),
  loading: false,
};


describe('Button Upload component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ButttonUpload {...props} />
    );

    expect(wrapper.find('.wrapcomp').name()).toEqual('Button');
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('renders as div', () => {
    const newProps = { ...props, as: 'div' };
    const wrapper = shallow(
      <ButttonUpload {...newProps} />
    );
    expect(wrapper.find('.wrapcomp').name()).toEqual('div');
    expect(wrapper.debug()).toMatchSnapshot();
  })

  it('upload is disabled', () => {
    const newProps = { ...props, disabled: true };
    const wrapper = shallow(
      <ButttonUpload {...newProps} />
    );
    expect(wrapper.find('.wrapcomp').prop('disabled')).toEqual(true);
    expect(wrapper.find('.file').prop('disabled')).toEqual(true);
  });
});
