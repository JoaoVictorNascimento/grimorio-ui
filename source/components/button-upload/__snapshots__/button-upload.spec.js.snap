// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Upload component renders as div 1`] = `
"<div className={[undefined]}>
  <div loading={false} disabled={false} iconLeft=\\"publish\\" className=\\"wrapcomp\\">
    Upload
    <input className=\\"file\\" type=\\"file\\" onChange={[Function: bound handleChange]} disabled={false} multiple={true} accept=\\".png, .jpg, .jpeg, .pdf\\" />
  </div>
  <div className=\\"holdTags\\" />
</div>"
`;

exports[`Button Upload component renders correctly 1`] = `
"<div className={[undefined]}>
  <Button loading={false} disabled={false} iconLeft=\\"publish\\" className=\\"wrapcomp\\" color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
    Upload
    <input className=\\"file\\" type=\\"file\\" onChange={[Function: bound handleChange]} disabled={false} multiple={true} accept=\\".png, .jpg, .jpeg, .pdf\\" />
  </Button>
  <div className=\\"holdTags\\" />
</div>"
`;
